src/ucs/memory/rcache_vfs.c

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2021 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucs/config/global_opts.h>
+#include <ucs/datastruct/queue.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/sys/math.h>
+#include <ucs/sys/string.h>
+#include <ucs/vfs/base/vfs_cb.h>
+#include <ucs/vfs/base/vfs_obj.h>
+#include "rcache_int.h"
+
+
+static void ucs_rcache_vfs_read_inv_q_length(void *obj,
+                                             ucs_string_buffer_t *strb,
+                                             void *arg_ptr, uint64_t arg_u64)
+{
+    ucs_rcache_t *rcache = obj;
+    size_t rcache_inv_q_length;
+
+    ucs_spin_lock(&rcache->lock);
+    rcache_inv_q_length = ucs_queue_length(&rcache->inv_q);
+    ucs_spin_unlock(&rcache->lock);
+
+    ucs_string_buffer_appendf(strb, "%zu\n", rcache_inv_q_length);
+}
+
+static void ucs_rcache_vfs_read_gc_list_length(void *obj,
+                                               ucs_string_buffer_t *strb,
+                                               void *arg_ptr, uint64_t arg_u64)
+{
+    ucs_rcache_t *rcache = obj;
+    unsigned long rcache_gc_list_length;
+
+    ucs_spin_lock(&rcache->lock);
+    rcache_gc_list_length = ucs_list_length(&rcache->gc_list);
+    ucs_spin_unlock(&rcache->lock);
+
+    ucs_string_buffer_appendf(strb, "%lu\n", rcache_gc_list_length);
+}
+
+static void ucs_rcache_vfs_show_primitive(void *obj, ucs_string_buffer_t *strb,
+                                          void *arg_ptr, uint64_t arg_u64)
+{
+    ucs_rcache_t *rcache = obj;
+
+    pthread_rwlock_rdlock(&rcache->pgt_lock);
+    ucs_vfs_show_primitive(obj, strb, arg_ptr, arg_u64);
+    pthread_rwlock_unlock(&rcache->pgt_lock);
+}
+
+static void ucs_rcache_vfs_init_regions_distribution(ucs_rcache_t *rcache)
+{
+    size_t num_bins = ucs_rcache_distribution_get_num_bins();
+    size_t i, threshold;
+    char memunits_str[32];
+
+    for (i = 0; i < num_bins; ++i) {
+        threshold = (i != num_bins - 1) ? UCS_RCACHE_STAT_MIN_POW2 << i :
+                                          UCS_MEMUNITS_INF;
+        ucs_vfs_obj_add_ro_file(rcache, ucs_rcache_vfs_show_primitive,
+                                &rcache->distribution[i], UCS_VFS_TYPE_SIZET,
+                                "regions_distribution/%s",
```

Comment:
"regions_distribution/%s/count"
"regions_distribution/%s/total_size"
(rcache->distribution needs to be a struct to have both count and size sum)